### ISSUE TITLE
strange behavior of skip-glob

### ISSUE THREAD

**Issue by matejcik** [timothycrosley/isort#950]
there's still a regression I've noticed in relation to #885, but it seems to warrant its own issue.

Assume an incorrectly sorted `foo/bar.py`.

With isort 4.3.10:
```
$ isort --check-only --recursive foo --skip-glob bar.py
Skipped 1 files
$ isort --check-only --recursive foo --skip-glob foo/bar.py
ERROR: /home/matejcik/a/foo/bar.py Imports are incorrectly sorted.
$ isort --check-only --recursive foo --skip-glob "*/foo/bar.py"
Skipped 1 files
$ isort --check-only --recursive foo --skip-glob "*/bar.py"
Skipped 1 files
```

with isort 4.3.11 and up (incl. 4.3.20):
```
$ isort --check-only --recursive foo --skip-glob bar.py
Skipped 1 files
$ isort --check-only --recursive foo --skip-glob foo/bar.py
ERROR: /home/matejcik/a/foo/bar.py Imports are incorrectly sorted.
$ isort --check-only --recursive foo --skip-glob "*/foo/bar.py"
ERROR: /home/matejcik/a/foo/bar.py Imports are incorrectly sorted.
$ isort --check-only --recursive foo --skip-glob "*/bar.py"
Skipped 1 files
```

I'm ... honestly not sure what to make of this. I'm not sure what the expected behavior is, but it's not this?

Pre-4.3.11, presumably the full absolute path was considered for matching, hence `bar.py` is OK because it means "any file named `bar.py` in any directory", but `foo/bar.py` is not OK because it's not the full path?

In 4.3.11 this was changed, but now instead of considering the path from CWD, you consider path from the `recursive` point? Which to me is very confusing.

I'd expect to match relative path from CWD.
Consider:
```
$ isort --check-only --recursive foo --skip foo/bar.py
Skipped 1 files
$ isort --check-only --recursive foo --skip-glob foo/bar.py
ERROR: /home/matejcik/a/foo/bar.py Imports are incorrectly sorted.
```
ISTM that with no wildcards, the two should behave identically.

**Comment by timothycrosley** [comment]
skip_glob no longer includes the full path, only the path being ran against, intentionally. This is due to things like */build/* skipping everything when ran on travis, when it was meant to refer to a local nested build directory

**Comment by matejcik** [comment]
I'm not arguing for using the full path though. Let me rephrase.

consider the following layout:
```
.
├── src
│   ├── a.py
│   └── messages
│       ├── a.py
│       └── b.py
└── tests
    ├── a.py
    └── messages
        ├── a.py
        └── b.py
```

typical isort invocation would be:
```
isort --recursive src tests
```
in order to ignore `tests/messages/*.py`, i would do:
```
isort --recursive src tests --skip-glob=tests/messages/*.py
```
that *doesn't work* though, because the globbed path starts at `tests`
the following does work:
```
isort --recursive src tests --skip-glob=messages/*.py
```
but that also skips `src/messages/*.py`, which is what I didn't want
